src/test/java/org/junitpioneer/jupiter/resource/ResourcesTests.java

```diff
@@ -0,0 +1,543 @@
+				ExecutionResults executionResults = PioneerTestKit.executeTestClass(ThrowOnNewRFCreateTestCases.class);
+				executionResults
+						.allEvents()
+						.debug()
+						.assertThatEvents()
+						.haveExactly(//
+							1, //
+							finished(//
+								throwable(//
+									instanceOf(ParameterResolutionException.class), //
+									message("Unable to create a resource from `" + ThrowOnRFCreateResourceFactory.class
+											+ "`"), //
+									cause(//
+										instanceOf(EXPECTED_THROW_ON_RF_CREATE_EXCEPTION.getClass()), //
+										message(EXPECTED_THROW_ON_RF_CREATE_EXCEPTION.getMessage())))));
+			}
+
+		}
+
+		@DisplayName("and the factory throws on ::close")
+		@Nested
+		class AndFactoryThrowsOnCloseTests {
+
+			@DisplayName("then the thrown exception is propagated")
+			@Test
+			void thenThrownExceptionIsPropagated() {
+				ExecutionResults executionResults = PioneerTestKit.executeTestClass(ThrowOnNewRFCloseTestCases.class);
+				executionResults
+						.allEvents()
+						.debug()
+						.assertThatEvents()
+						.haveExactly(//
+							1, //
+							finished(//
+								throwable(//
+									instanceOf(EXPECTED_THROW_ON_RF_CLOSE_EXCEPTION.getClass()), //
+									message(EXPECTED_THROW_ON_RF_CLOSE_EXCEPTION.getMessage()))));
+			}
+
+		}
+
+		@DisplayName("and a resource is created")
+		@Nested
+		class AndResourceIsCreatedTests {
+
+			@DisplayName("and the resource throws on ::get")
+			@Nested
+			class AndResourceThrowsOnGetTests {
+
+				@DisplayName("then the thrown exception is wrapped and propagated")
+				@Test
+				void thenThrownExceptionIsWrappedAndPropagated() {
+					ExecutionResults executionResults = PioneerTestKit.executeTestClass(ThrowOnNewRGetTestCases.class);
+					executionResults
+							.allEvents()
+							.debug()
+							.assertThatEvents()
+							.haveExactly(//
+								1, //
+								finished(//
+									throwable(//
+										instanceOf(ParameterResolutionException.class), //
+										message("Unable to get the contents of the resource created by `"
+												+ ThrowOnRGetResourceFactory.class + "`"), //
+										cause(//
+											instanceOf(EXPECTED_THROW_ON_R_GET_EXCEPTION.getClass()), //
+											message(EXPECTED_THROW_ON_R_GET_EXCEPTION.getMessage())))));
+				}
+
+			}
+
+			@DisplayName("and the resource throws on ::close")
+			@Nested
+			class AndResourceThrowsOnCloseTests {
+
+				@DisplayName("then the thrown exception is propagated")
+				@Test
+				void thenThrownExceptionIsWrappedAndPropagated() {
+					ExecutionResults executionResults = PioneerTestKit
+							.executeTestClass(ThrowOnNewRCloseTestCases.class);
+					executionResults
+							.allEvents()
+							.debug()
+							.assertThatEvents()
+							.haveExactly(//
+								1, //
+								finished(//
+									throwable(//
+										instanceOf(EXPECTED_THROW_ON_R_CLOSE_EXCEPTION.getClass()), //
+										message(EXPECTED_THROW_ON_R_CLOSE_EXCEPTION.getMessage()))));
+				}
+
+			}
+
```

Comment:
More tests:

* assert that for every `@New`, `ResourceFactory::create` is called
* assert that if `ResourceFactory::create` returns `null` or `Resource<null>` a proper exception is thrown (to avoid NPEs that would be confusing to users)

Reply:
Thanks for the feedback! Partially done: We now assert that if `ResourceFactory::create` returns `null` then a proper exception is thrown.

For my reference, the remaining tasks are:
- Assert that if `ResourceFactory::create` returns `Resource<null>` then a proper exception is thrown.
- Assert that for every `@New`, `ResourceFactory::create` is called.

Reply:
@nipafx This comment is now addressed! I trust you'll let me know if there's anything else that needs to be followed up here.